src/Umbraco.Core/EmbeddedResources/BlockGrid/items.cshtml

```diff
@@ -17,9 +20,17 @@
             style=" --umb-block-grid--item-column-span: @item.ColumnSpan; --umb-block-grid--item-row-span: @item.RowSpan; ">
             @{
                 var partialViewName = "blockgrid/Components/" + item.Content.ContentType.Alias;
+
                 try
                 {
-                    @await Html.PartialAsync(partialViewName, item)
+                    if (_selector.SelectComponent(item.Content.ContentType.Alias) is not null)
+                    {
+                        @await Component.InvokeAsync(item.Content.ContentType.Alias, item.Content)
```

Comment:
I think the model should stay the same as the partial to avoid confusion.

```suggestion
                        @await Component.InvokeAsync(item.Content.ContentType.Alias, item)
```

Reply:
I have updated this, so `BlockGridItem` model is pass in to component and not just `IPublishedContent` Content property.
Furthermore I noticed a few warnings at XML comments for `rowSpan` and `columnSpan`, but the ctor's didn't have any overload taking these parameters.